### PR TITLE
[Vaults] Fix: filterAndCropContentNodesByView parents computation

### PR DESCRIPTION
**Pull request by philipperolet**
Description
---
Former version of `filterAndCropContentNodesByView` correctly excluded nodes, but incorrectly filtered parentsInternalIds: some parents that were in the view were excluded from it

This PR fixes it

Risk
---
na

Deploy
---
front

### CODE REVIEW

File: front/lib/api/data_source_view.ts [hunk 1]
```diff
@@ -18,14 +18,22 @@ export function filterAndCropContentNodesByView(
...
+    // At the first parent that is in the view, we know the content node is in the view
     const indexToSplit = parentInternalIds.findIndex((p) =>
       dataSourceView.parentsIn?.includes(p)
     );
     const isInView = !viewHasParents || indexToSplit !== -1;
 
     if (isInView) {
-      const parentIdsInView = parentInternalIds.slice(0, indexToSplit);
+      // for parents, we include all those up to the last one in the view
+      // (or all of them if the view has no parents)
```
Comment (flvndvd) [inline front/lib/api/data_source_view.ts:29]:
```suggestion
      // For parents, we include all those up to the last one in the view
      // (or all of them if the view has no parents).
```

File: front/lib/api/data_source_view.ts [hunk 2]
```diff
@@ -18,14 +18,22 @@ export function filterAndCropContentNodesByView(
       return null;
     }
 
-    // We stop at the first parent that is in the view.
+    // At the first parent that is in the view, we know the content node is in the view
     const indexToSplit = parentInternalIds.findIndex((p) =>
```
Comment (flvndvd) [inline front/lib/api/data_source_view.ts:22]: We should actually update this to go all the way up to the oldest ancestor that is in the view, no? 🤔

Comment (philipperolet) [reply front/lib/api/data_source_view.ts:22]: As per irl :+1:

File: front/lib/api/data_source_view.ts [hunk 3]
```diff
@@ -18,14 +18,20 @@ export function filterAndCropContentNodesByView(
       return null;
     }
 
-    // We stop at the first parent that is in the view.
-    const indexToSplit = parentInternalIds.findIndex((p) =>
+    // check that the node or at least a parent is in the view we know the
+    // content node is in the view
+    // for parents, we include all those up to the last one in the view
+    // (or all of them if the view has no parents)
```
Comment (flvndvd) [inline front/lib/api/data_source_view.ts:24]:
```suggestion
    // Ensure that the node, or at least one of its ancestors, is within the view.
    // For ancestor nodes, include all of them up to the last one that resides within the view,
    // or include all ancestors if the view has no parent nodes.
```

**Review by flvndvd — APPROVED** [review]
LGTM just left a nit on the code comment.

**Pull request merged**